transaction/core/src/account_keys.rs

```diff
@@ -13,37 +13,59 @@
 #![allow(non_snake_case)]
 
 use crate::{domain_separators::SUBADDRESS_DOMAIN_TAG, view_key::ViewKey};
-use alloc::string::{String, ToString};
-use blake2::{Blake2b, Digest};
+
 use core::{
     cmp::Ordering,
     fmt,
     hash::{Hash, Hasher},
 };
-use curve25519_dalek::scalar::Scalar;
+
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
 use mc_crypto_digestible::Digestible;
 use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
 use mc_util_from_random::FromRandom;
+
+use blake2::{Blake2b, Digest};
+use curve25519_dalek::scalar::Scalar;
 use prost::Message;
 use rand_core::{CryptoRng, RngCore};
-use serde::{Deserialize, Serialize};
 
 /// An account's "default address" is its zero^th subaddress.
 pub const DEFAULT_SUBADDRESS_INDEX: u64 = 0;
 
+/// An account's version number. If breaking changes are made to the interpretation
+/// of account keys or public addresses, this should be incremented so that old clients
+/// will fail fast on data they shouldn't handle.
+pub const CURRENT_ACCOUNT_KEY_VERSION: u32 = 0;
+
 /// A MobileCoin user's public subaddress.
-#[derive(
-    PartialEq, Eq, PartialOrd, Ord, Hash, Default, Debug, Serialize, Deserialize, Clone, Digestible,
-)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Message, Clone, Digestible)]
 pub struct PublicAddress {
     /// The user's public subaddress view key 'C'.
-    view_public_key: RistrettoPublic,
+    #[prost(message, required, tag = "1")]
+    pub view_public_key: RistrettoPublic,
 
     /// The user's public subaddress spend key `D`.
-    spend_public_key: RistrettoPublic,
+    #[prost(message, required, tag = "2")]
+    pub spend_public_key: RistrettoPublic,
 
     /// Fog Url, if the user has a fog service
-    fog_url: Option<String>,
+    /// Should be parseable as mc_util_uri::FogUri.
+    #[prost(string, tag = "3")]
+    pub fog_url: String,
+
+    /// A signature with the user's view_private_key over the fog authority key fingerprint.
+    /// Empty if no fog for this address
+    #[prost(bytes, tag = "4")]
+    pub fog_authority_sig: Vec<u8>,
```

Comment:
Could you add an accessor that returns `Option<&[u8]>` in the same vein as `fog_url(&self) -> Option<&str>`?